hooks/tk-multi-publish2/ingest/collector.py

```diff
@@ -328,37 +328,50 @@ def process_file(self, settings, parent_item, path):
         # make sure we have snapshot_type field in all the items!
         # this is to make sure that on publish we retain this field to figure out asset creation is needed or not.
         for file_item in file_items:
-            fields = file_item.properties["fields"]
+            # create the defaults on the item, if we didn't already get them from the manifest.
+            self.create_ingest_default_fields(settings, file_item)
 
-            item_info = self._get_item_type_info(settings, file_item.type)
+        return file_items
 
-            if "snapshot_type" not in fields:
-                fields["snapshot_type"] = item_info["default_snapshot_type"]
-                # CDL files should always be published as Asset entity with nuke_avidgrade asset_type
-                # this is to match organic, and also for Avid grade lookup on shotgun
-                # this logic has been moved to _get_item_type_info by defining default_snapshot_type for each item type
-                # if file_item.type == "file.cdl":
-                #     fields["snapshot_type"] = "nuke_avidgrade"
+    def create_ingest_default_fields(self, settings, item):
```

Comment:
Why wouldn't you do all of this as a method override of `_resolve_item_fields()`? Seems like the more appropriate place...plus it will run with the updated settings should they change.

Reply:
Yep, will move this to `_resolve_item_fields()`.